src/natcap/invest/habitat_quality.py

```diff
@@ -477,68 +525,76 @@ def execute(args):
 
             # get the threat raster for the specific threat
             threat_raster_path = threat_path_dict['threat' + lulc_key][threat]
-            LOGGER.info('threat_raster_path %s', threat_raster_path)
+            # if threat path is None then must be in Base scenario where
+            # threats are not required.
             if threat_raster_path is None:
-                LOGGER.info(
-                    'The threat raster for %s could not be found for the land '
-                    'cover %s. Skipping Habitat Quality calculation for this '
-                    'land cover.' % (threat, lulc_key))
+                LOGGER.warning(
+                    f'The threat raster for {threat} could not be found for'
+                    f' the land cover {lulc_key}. Skipping Habitat Quality'
+                    ' calculation for this land cover.')
                 exit_landcover = True
                 break
 
-            # need the pixel size for the threat raster so we can create
-            # an appropriate kernel for convolution
-            threat_pixel_size = pygeoprocessing.get_raster_info(
-                threat_raster_path)['pixel_size']
-            # pixel size tuple could have negative value
-            mean_threat_pixel_size = (
-                abs(threat_pixel_size[0]) + abs(threat_pixel_size[1]))/2.0
-
-            # convert max distance (given in KM) to meters
-            max_dist_m = threat_data['MAX_DIST'] * 1000.0
-
-            # convert max distance from meters to the number of pixels that
-            # represents on the raster
-            max_dist_pixel = max_dist_m / mean_threat_pixel_size
-            LOGGER.debug('Max distance in pixels: %f', max_dist_pixel)
-
-            # blur the threat raster based on the effect of the threat over
-            # distance
-            decay_type = threat_data['DECAY']
             kernel_path = os.path.join(
-                kernel_dir, 'kernel_%s%s%s.tif' % (threat, lulc_key, suffix))
-            if decay_type == 'linear':
-                make_linear_decay_kernel_path(max_dist_pixel, kernel_path)
-            elif decay_type == 'exponential':
-                utils.exponential_decay_kernel_raster(
-                    max_dist_pixel, kernel_path)
-            else:
-                raise ValueError(
-                    "Unknown type of decay in biophysical table, should be "
-                    "either 'linear' or 'exponential'. Input was %s for threat"
-                    " %s." % (decay_type, threat))
+                kernel_dir, f'kernel_{threat}{lulc_key}{file_suffix}.tif')
+
+            decay_type = threat_data['decay']
+
+            create_kernel_task = task_graph.add_task(
+                _create_decay_kernel,
+                args=((threat_raster_path, 1), kernel_path, decay_type,
+                      threat_data['max_dist']),
+                target_path_list=[kernel_path],
+                dependent_task_list=[*bound_threat_values_tasks],
```

Comment:
`bound_threat_values_tasks` is already a list right? so you can set it directly to `dependent_task_list`?